### PR TITLE
feat: `azure-storage-blob` driver

### PR DESCRIPTION
**Pull request by itpropro**

### CODE REVIEW

File: src/drivers/azure-storage-blob.ts [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    },
+    async setItem(key, value) {
+      await getContainerClient()
+        .getBlockBlobClient(key)
+        .upload(value, Buffer.byteLength(value));
+      return;
+    },
+    async removeItem(key) {
+      await getContainerClient().getBlockBlobClient(key).delete();
+      return;
```
Comment (danielroe) [inline src/drivers/azure-storage-blob.ts:109]:
```suggestion
```

File: src/drivers/azure-storage-blob.ts [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+        return (await streamToBuffer(blob.readableStreamBody)).toString();
+      } catch {
+        return null;
+      }
+    },
+    async setItem(key, value) {
+      await getContainerClient()
+        .getBlockBlobClient(key)
+        .upload(value, Buffer.byteLength(value));
+      return;
```
Comment (danielroe) [inline src/drivers/azure-storage-blob.ts:105]:
```suggestion
```

File: src/drivers/azure-storage-blob.ts [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+        await Promise.all(
+          page.segment.blobItems.map(
+            async (blob) =>
+              await getContainerClient().deleteBlob(blob.name, {
+                deleteSnapshots: "include",
+              })
+          )
+        );
+      }
+      return;
```
Comment (danielroe) [inline src/drivers/azure-storage-blob.ts:147]:
```suggestion
```

**Review by danielroe — APPROVED** [review]

**Review by pi0 — APPROVED** [review]
Nice work & LGTM. I think next steps we can enable raw support for this driver (which is probably it's main usesecause i guess)

**Comment by itpropro** [comment]
> Nice work & LGTM. I think next steps we can enable raw support for this driver (which is probably it's main usesecause i guess)

Yes, raw support would be a great next step and the driver would be perfectly fitted for that :)

**Pull request merged**